hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java

```diff
@@ -334,31 +360,6 @@ public String extractPattern(TemporalUnit unit) {
 		}
 	}
 
-	/**
-	 * MySQL doesn't have a real {@link java.sql.Types#BOOLEAN}
-	 * type, so...
-	 */
-	@Override
-	public String castPattern(CastType from, CastType to) {
-		switch (to) {
-			case BOOLEAN:
-				switch (from) {
-					case STRING:
-//						return "if(?1 rlike '^(t|f|true|false)$', ?1 like 't%', null)";
-						return "if(lower(?1) in('t','f','true','false'), ?1 like 't%', null)";
-					case LONG:
-					case INTEGER:
-						return "(?1<>0)";
-				}
-			case STRING:
-				if (from == BOOLEAN) {
-					return "if(?1,'true','false')";
```

Comment:
same comment here: this `if()` function is much nicer than a `case`\`when`

Reply:
Passing null would produce the else-result, so this wouldn't be compatible with other implementations. Thus it needs case when.